projects/01-url-shortner/amir-jahandar/urlshortner/shorturl/views.py

```diff
@@ -0,0 +1,93 @@
+import profile
+from django.shortcuts import redirect, render
+from django.http import HttpResponse
+from .models import ShortUrl, Profile
+from .forms import CreateNewShortUrl, SignUpForm
+from django.contrib.auth import login as dj_login , authenticate, logout as dj_logout
+from datetime import datetime
+import random, string
+
+
+def home(request):
+    if request.user.is_authenticated:
+        user_urls = ShortUrl.objects.filter(profile__user__id = request.user.id)
+        return render(request, 'home.html', {'msg':'Login successfuly!', 'user_urls':user_urls})
+    return render(request, 'home.html')
+
+
+def createshorturl(request):
+    if request.method == 'POST':
+        form = CreateNewShortUrl(request.POST)
+        if form.is_valid():
+            originalurl = form.cleaned_data['original_url']
+            random_chars_list = list(string.ascii_letters)
```

Comment:
ascii_letters doesn't contain digits. In order to generate more urls, you can use digits too.

Reply:
now generate short url combine digits too.